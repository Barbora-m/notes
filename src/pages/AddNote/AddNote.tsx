import { useCallback, useState } from "react";
import { Alert, Button, CardBody, Col, Input, Row } from "reactstrap";
import styled from "styled-components";
import { useHistory } from "react-router-dom";
import { useTranslation } from "react-i18next";

import { client } from "../../client";

export const AddNote = () => {
  const { push } = useHistory();
  const { t } = useTranslation();

  const [noteText, setNoteText] = useState("");
  const [isError, setIsError] = useState(false);

  const handleNoteChange = useCallback((event) => {
    setNoteText(event.target.value);
  }, []);

  const handleBackButtonClick = useCallback(() => {
    push("/notes");
  }, [push]);

  const handleNoteAdd = useCallback(() => {
    try {
      void client.post("/", {
        content: { noteText },
      });

      push("/notes");
      setNoteText("");
    } catch (error) {
      setIsError(true);
    }
  }, [noteText, push]);

  return (
    <CardBody>
      {isError && <Alert color="danger">t{"noteError"}</Alert>}
      <Input
        onChange={handleNoteChange}
        placeholder={t("addNoteInputPlaceholder")}
        value={noteText}
      />

      <Row>
        <ButtonsWrapper>
          <BackButton size="sm" onClick={handleBackButtonClick}>
            {t("back")}
          </BackButton>
          <Button
            color="success"
            disabled={!noteText}
            onClick={handleNoteAdd}
            size="sm"
            type="submit"
          >
            {t("addNote")}
          </Button>
        </ButtonsWrapper>
      </Row>
    </CardBody>
  );
};

const ButtonsWrapper = styled(Col)`
  display: flex;
  justify-content: space-between;
  margin-top: 25px;
`;

const BackButton = styled(Button)`
  margin-right: 5px;
`;
