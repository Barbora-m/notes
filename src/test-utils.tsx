import { ReactElement } from "react";
import { I18nextProvider } from "react-i18next";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import { i18n } from "./i18n";

export const renderWithProviders = (component: ReactElement) =>
  renderer.create(
    <I18nextProvider i18n={i18n}>
      <MemoryRouter>{component}</MemoryRouter>
    </I18nextProvider>
  );
