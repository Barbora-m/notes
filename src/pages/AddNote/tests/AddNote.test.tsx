import { renderWithProviders } from "../../../test-utils";
import { AddNote } from "../AddNote";

test("AddNote", () => {
  expect(renderWithProviders(<AddNote />)).toMatchSnapshot();
});
