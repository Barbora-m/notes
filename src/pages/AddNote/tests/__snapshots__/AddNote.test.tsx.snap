// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNote 1`] = `
<div
  className="card-body"
>
  <input
    className="form-control"
    onChange={[Function]}
    placeholder="Please enter your note here..."
    type="text"
    value=""
  />
  <div
    className="row"
  >
    <div
      className="sc-bdfBwQ bgajQq col"
    >
      <button
        aria-label={null}
        className="sc-gsTCUz ca-dqFr btn btn-secondary btn-sm"
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
      <button
        aria-label={null}
        className="btn btn-success btn-sm disabled"
        disabled={true}
        onClick={[Function]}
        type="submit"
      >
        Add note
      </button>
    </div>
  </div>
</div>
`;
